totalRP3/Modules/NamePlates/NamePlates_Platynator.lua

```diff
@@ -0,0 +1,168 @@
+-- Copyright The Total RP 3 Authors
+-- SPDX-License-Identifier: Apache-2.0
+
+
+-- plusmouse has kindly provided an API to modify unit names. See https://github.com/TheMouseNest/Platynator/commit/2446f3461c598b93d7e63f4ac175a5ed0e08d119
+-- We do not have access to the FontString itself. Long names may be truncated, Platynator users will have to adjust the settings themselves.
+-- Custom name colors and icons need to be wrapped together in escape sequences.
+-- Custom health colors will not be supported.
+
+
+local TRP3_API = select(2, ...);
+local TRP3_NamePlates = TRP3_NamePlates;
+local TRP3_NamePlatesUtil = TRP3_NamePlatesUtil;
+local L = TRP3_API.loc;
+
+
+local canaccessvalue = canaccessvalue or function(_) return true; end;
+
+
+local DisplayManager = {};
+do
+	function DisplayManager:SetUnitText(unitToken, creatureName, guildName)
+		self.overrideFunc(unitToken, creatureName, guildName)
+	end
+
+	function DisplayManager:Initialize()
+		if not C_AddOns.IsAddOnLoaded("Platynator") then
+			return false;
+		end
+
+		local requiredAPIs = {
+			overrideFunc = "Platynator.API.SetUnitTextOverride",
+		};
+
+		local function GetGlobalObject(objNameKey)
+			--Get an object via string "Name.Key1.Key2..."
+			local obj = _G;
+			for k in string.gmatch(objNameKey, "[_%w]+") do
+				obj = obj[k];
+				if not obj then
+					return
+				end
+			end
+			return obj
+		end
+
+		for k, objNameKey in pairs(requiredAPIs) do
+			local getterFunc = GetGlobalObject(objNameKey);
+			if getterFunc then
+				self[k] = getterFunc;
+			else
+				return false;
+			end
+		end
+
+		return true
+	end
+end
```

Comment:
This entire thing seems rather convoluted - I don't see why we're parsing a string to resolve one function, and then assigning it to a `self` field with a generic `overrideFunc` name.

Get rid of it and just do the following:

1. Adjust checks in OnModuleInitialize to check for required APIs.
2. Directly call SetUnitTextOverride in UpdateNamePlate.